src/app/features/course-info/course-info.component.ts

```diff
@@ -1,10 +1,19 @@
-import { Component } from '@angular/core';
+import { Component, Input } from '@angular/core';
+
+interface CourseModel {
```

Comment:
You are absolutely right to mark the shape of the 'course' entity this way. However, I saw this interface being declared on multiple component files. You need to declare it only once and distribute it via export and imports. Best create a new folder for accommodating all model related files in shared folder and put this interface in a separate file.